CommonTools/RecoAlgos/plugins/JetConstituentSelector.cc

```diff
@@ -20,23 +20,29 @@
 #include "DataFormats/JetReco/interface/GenJet.h"
 #include "DataFormats/PatCandidates/interface/Jet.h"
 #include "DataFormats/PatCandidates/interface/PackedCandidate.h"
+#include "DataFormats/PatCandidates/interface/PackedGenParticle.h"
 #include "DataFormats/JetReco/interface/CaloJet.h"
 #include "FWCore/Framework/interface/stream/EDProducer.h"
 #include "FWCore/Framework/interface/MakerMacros.h"
 #include "FWCore/Framework/interface/Event.h"
 #include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
 #include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
 
+
+
+
 template <class T, typename C = std::vector<typename T::ConstituentTypeFwdPtr>>
 class JetConstituentSelector : public edm::stream::EDProducer<> {
 public:
 
   using JetsOutput = std::vector<T>;
   using ConstituentsOutput = C;
-
+  using ValueType = typename C::value_type;
+  
   JetConstituentSelector(edm::ParameterSet const& params) :
     srcToken_{consumes<edm::View<T>>(params.getParameter<edm::InputTag>("src"))},
-    selector_{params.getParameter<std::string>("cut")}
+    selector_{params.getParameter<std::string>("cut")},
+    unpackAK8_{params.existsAs<bool>("unpackAK8") ? params.getParameter<bool>("unpackAK8") : false}
```

Comment:
Please, add a "false" as default in the fillDescriptions() method, and get rid of the ```existsAs``` here

Reply:
Won't that affect existing workflows though? If they don't have the parameter they will need to add it. (Or is adding "false" as a default sufficient?)

Reply:
Adding "false" as default in the parameterSetDescription insures that whenever the parameter is missing the default value is assigned.

Which is the same as you are doing now, but a bit cleaner from the point of view of the code.

Reply:
existsAs is strongly discouraged and exceptions to use it are still possible only for plugins.
This is not the case for this producer.
So, I think that fillDescriptions should be added.

Reply:
OK I will fix it. It looks like there are some matrix failures for some reason, I will fix those too.